src/chain/data_chain.rs

```diff
@@ -315,7 +315,7 @@ impl DataChain {
         self.chain
             .iter()
             .cloned()
-            .filter(|x| !x.identifier().is_link() && x.valid)
+            .filter(|x| !x.identifier().is_block() && x.valid)
```

Comment:
Ah, that little exclamation mark is what I was missing! Then this change isn't needed.